TShockAPI/TShockAPI.csproj

```
@@ -1,242 +1,71 @@
-    <AssemblyName>TShockAPI</AssemblyName>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <FileAlignment>512</FileAlignment>
-    <IsWebBootstrapper>false</IsWebBootstrapper>
-    <PublishUrl>publish\</PublishUrl>
-    <Install>true</Install>
-    <InstallFrom>Disk</InstallFrom>
-    <UpdateEnabled>false</UpdateEnabled>
-    <UpdateMode>Foreground</UpdateMode>
-    <UpdateInterval>7</UpdateInterval>
-    <UpdateIntervalUnits>Days</UpdateIntervalUnits>
-    <UpdatePeriodically>false</UpdatePeriodically>
-    <UpdateRequired>false</UpdateRequired>
-    <MapFileExtensions>true</MapFileExtensions>
-    <ApplicationRevision>0</ApplicationRevision>
-    <ApplicationVersion>1.0.0.%2a</ApplicationVersion>
-    <UseApplicationTrust>false</UseApplicationTrust>
-    <BootstrapperEnabled>true</BootstrapperEnabled>
-    <TargetFrameworkProfile />
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-    <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
-    <Optimize>false</Optimize>
-    <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <DocumentationFile>bin\Debug\TShockAPI.XML</DocumentationFile>
-    <PlatformTarget>x86</PlatformTarget>
-    <Prefer32Bit>false</Prefer32Bit>
-  </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE;COMPAT_SIGS</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+    <TargetFramework>net5.0</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <DocumentationFile>bin\Release\TShockAPI.XML</DocumentationFile>
-    <PlatformTarget>x86</PlatformTarget>
-    <Prefer32Bit>false</Prefer32Bit>
+    <GenerateDocumentationFile>True</GenerateDocumentationFile>
+    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
+    <PackageLicenseExpression>GPL-3.0-or-later</PackageLicenseExpression>
+    <!-- 
+      Version information for an assembly consists of the following four values:
+
+          Major Version
+          Minor Version 
+          Build Number
+          Starting in version 4.2.5, we are no longer including the fourth decimal
+          location, which previously held the date and time.
+
+      Also, be sure to release on github with the exact assembly version tag as below
+      so that the update manager works correctly (via the Github releases api and mimic)
+    -->
+    <Version>4.5.7</Version>
+    <AssemblyVersion>4.5.7</AssemblyVersion>
+    <FileVersion>4.5.7</FileVersion>
+    <AssemblyTitle>TShock for Terraria</AssemblyTitle>
+    <Company>Re-Logic, Pryaxis &amp; TShock Contributors</Company>
```

```suggestion
    <Company>Pryaxis &amp; TShock Contributors</Company>
```

I think we can drop RL from this because we aren't packing their stuff here anymore. I think this copyright was vestigial from back in the "we are shipping their source code with TShock" days.

---

that one is up to you, so sure? this was just moved from AssemblyInfo.cs

---

@DeathCradle it's your branch :P